generator/templates/base_struct_function.m

```diff
@@ -0,0 +1,66 @@
+{#- To avoid code duplication was crated this parent file, which contain common part used in:
+    "templates/functions/template.m" and "templates/structs/template.m". -#}
+{% include 'copyright.txt' %}
+{%- block imports %}
+#import "{{name}}.h"
+#import "NSMutableDictionary+Store.h"
+{%- for import in imports %}
+#import "{{import}}.h"
+{%- endfor %}
+{%- endblock %}
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation {{name}}
+{% block constructors %}
+{%- for c in constructors %}
+- (instancetype)initWith{{c.init}} {
```

Comment:
When creating an `init` that uses a deprecated type (for example, `SDLSeatControlData` which uses the deprecated `SDLSupportedSeat`) we need to wrap the `init` in

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"

#pragma clang diagnostic pop
```

Reply:
@joeljfischer all parameter should be wrapped together?
```objc
...
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
/**
 * Predefined or dynamically created screen layout. Currently only predefined screen layouts are defined.
 * {"default_value": null, "max_length": 500, "min_length": 1}
 */
@property (strong, nonatomic) NSString *displayLayout;

/**
 * @since SDL 5.0.0
 */
@property (nullable, strong, nonatomic) SDLTemplateColorScheme *dayColorScheme;

/**
 * @since SDL 5.0.0
 */
@property (nullable, strong, nonatomic) SDLTemplateColorScheme *nightColorScheme;

#pragma clang diagnostic pop
...
```

Or each parameter should be wrapped separately?
```objc
...
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
/**
 * Predefined or dynamically created screen layout. Currently only predefined screen layouts are defined.
 * {"default_value": null, "max_length": 500, "min_length": 1}
 */
@property (strong, nonatomic) NSString *displayLayout;
#pragma clang diagnostic pop

#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
/**
 * @since SDL 5.0.0
 */
@property (nullable, strong, nonatomic) SDLTemplateColorScheme *dayColorScheme;
#pragma clang diagnostic pop

#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
/**
 * @since SDL 5.0.0
 */
@property (nullable, strong, nonatomic) SDLTemplateColorScheme *nightColorScheme;
#pragma clang diagnostic pop
...
```

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/97fc74359dc0ac2b615be9fb615416edb22becd9

Reply:
For some reason, I'm seeing deprecations not being wrapped in initialized parameters. The easiest way to solve this is that instead of wrapping just the `initWith:`, to wrap the entire method. If you're going to have to wrap the `initWith`, you have to wrap the entire method anyway.

Reply:
@joeljfischer I don't get your question, currently generated code looks like as listed below.
How do you supposes it to looks like?
```objc
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
- (instancetype)initWithIdParam:(SDLSupportedSeat)idParam {
#pragma clang diagnostic pop
...
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
- (instancetype)initWithIdParam:(SDLSupportedSeat)idParam heatingEnabled:(BOOL)heatingEnabled coolingEnabled:(BOOL)coolingEnabled heatingLevel:(nullable NSNumber<SDLUInt> *)heatingLevel coolingLevel:(nullable NSNumber<SDLUInt> *)coolingLevel horizontalPosition:(nullable NSNumber<SDLUInt> *)horizontalPosition verticalPosition:(nullable NSNumber<SDLUInt> *)verticalPosition frontVerticalPosition:(nullable NSNumber<SDLUInt> *)frontVerticalPosition backVerticalPosition:(nullable NSNumber<SDLUInt> *)backVerticalPosition backTiltAngle:(nullable NSNumber<SDLUInt> *)backTiltAngle headSupportHorizontalPosition:(nullable NSNumber<SDLUInt> *)headSupportHorizontalPosition headSupportVerticalPosition:(nullable NSNumber<SDLUInt> *)headSupportVerticalPosition massageEnabled:(BOOL)massageEnabled massageMode:(nullable NSArray<SDLMassageModeData *> *)massageMode massageCushionFirmness:(nullable NSArray<SDLMassageCushionFirmness *> *)massageCushionFirmness memory:(nullable SDLSeatMemoryAction *)memory {
#pragma clang diagnostic pop
...
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
- (void)setIdParam:(SDLSupportedSeat)idParam {
#pragma clang diagnostic pop
...
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
- (SDLSupportedSeat)idParam {
#pragma clang diagnostic pop
...
```

Reply:
```objc
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
- (instancetype)initWithIdParam:(SDLSupportedSeat)idParam {
    self = [super init];
    if (!self) {
        return nil;
    }
    self.idParam = idParam;
    return self;
}
#pragma clang diagnostic pop
```

Reply:
Yes, like that. The `self.idParam = idParam` was not being wrapped previously, and it was causing a warning.

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/8d538469711fe0690157629c5f8d0c361b29cdff